src/control/lib/spdk/spdk.go

```diff
@@ -126,7 +130,13 @@ func (e *Env) InitSPDKEnv(opts EnvOptions) error {
 		return Rc2err("spdk_env_init()", rc)
 	}
 
-	if !opts.EnableVMD {
+	log.Debugf("spdk init c opts: %+v", cOpts)
+
+	if toFree != nil {
+		defer C.free(unsafe.Pointer(*toFree))
+	}
```

Comment:
A little confused by moving this defer later in the routine - doesn't this leak toFree now if we exit early for an error?